### PR TITLE
PFMENG-203 : Add vault workload identity hooks and tests

### PR DESCRIPTION
**Pull request by smoneyan**

### CODE REVIEW

File: hooks/terraform-pre-plan [hunk 1]
```diff
@@ -89,3 +89,30 @@ if [ -z ${TFC_GCP_PLAN_SERVICE_ACCOUNT_EMAIL+x} ] && [ -z ${TFC_GCP_RUN_SERVICE_
...
+  echo "Skipping VAULT Auth";
+  else
+    set +u
+    VAULT_ROLE="${TFC_VAULT_PLAN_ROLE:-$TFC_VAULT_RUN_ROLE}"
+    set -u
+    echo "Preparing VAULT provider auth..."
+
+    jq -n -r --arg JWT "$TFC_WORKLOAD_IDENTITY_TOKEN" --arg ROLE "$VAULT_ROLE" '{
+      "jwt": $JWT, 
+      "role": $ROLE }' > payload.json
```
Comment (lawliet89) [inline hooks/terraform-pre-plan:106]: Maybe let's write this to `~/.vault-payload.json`?

**Review by lawliet89 — COMMENTED** [review]
Dockerfile requires some changes too, if you refer to the doc.

File: hooks/terraform-pre-plan [hunk 2]
```diff
@@ -89,3 +89,30 @@ if [ -z ${TFC_GCP_PLAN_SERVICE_ACCOUNT_EMAIL+x} ] && [ -z ${TFC_GCP_RUN_SERVICE_
...
+
+    VAULT_TOKEN=$(curl --silent --show-error "$VAULT_ADDR/v1/auth/jwt/login" \
+      --header "X-Vault-Namespace: $VAULT_NAMESPACE" \
+      --request POST \
+      --data @payload.json | jq -r ".auth.client_token")
+
+    echo "${VAULT_TOKEN}" > ~/.vault-token
```
Comment (lawliet89) [inline hooks/terraform-pre-plan:113]: Add a `DEBUG` output similar to the other auth methods to run `vault token lookup`

File: hooks/terraform-pre-plan (hunk 1 above)
Comment (smoneyan) [reply hooks/terraform-pre-plan]: done

File: hooks/terraform-pre-plan (hunk 2 above)
Comment (smoneyan) [reply hooks/terraform-pre-plan:113]: done

**Comment by smoneyan** [comment]
> Dockerfile requires some changes too, if you refer to the doc.

Hmm. These docker changes are already part of the Dockerfile already. Nothing specific for vault. 
image

File: hooks/terraform-pre-plan (hunk 2 above)
Comment (lawliet89) [reply hooks/terraform-pre-plan:113]: You left out the debug in plan.

Comment (smoneyan) [reply hooks/terraform-pre-plan:113]: 🤦  .. added it now.

**Review by lawliet89 — APPROVED** [review]

File: hooks/terraform-pre-plan (hunk 2 above)
Comment (niroz89) [reply hooks/terraform-pre-plan:113]: makes me eager to try out bats.

**Pull request merged**